src/telemetry.rs

```diff
@@ -37,8 +38,19 @@ pub fn setup_logging() {
     tracing_subscriber::registry().with(fmt_layer()).init();
 }
 
+fn is_local_dev() -> bool {
```

Comment:
I was thinking something like https://github.com/dtolnay/isatty to detect if we should use color. Probably not this exact crate since its archive though, but detecting if we should use colors or not is fairly standard in linux programs

Reply:
Thanks for clarifying. I'll explore the crates.

Reply:
> We may also want to adapt to Istio log format eventually

By the way, IIUC, does this mean that we need to create a custom formatter to replace the current event formatter in https://github.com/istio/ztunnel/pull/476/files#diff-92d65ac8fde000b1d3ee1fda746b57e2586a6e7824e52256a9961016a10779deR53 under certain conditions?

Reply:
> I was thinking something like https://github.com/dtolnay/isatty to detect if we should use color. Probably not this exact crate since its archive though, but detecting if we should use colors or not is fairly standard in linux programs

Revised to `atty` to detect if we should use color.



> Yeah I think so, but not urgent. Might be a lot of work

Yes, I've looked into it, and it seems like there are many things that need to be dealt with, and some of the work may need to be finished upstream. I'll just leave the issue open and look back at it when I have time.